### PR TITLE
feat!: add target flag

### LINKED ISSUE #5: Add macOS and Windows support

### PR DESCRIPTION
**Pull request by SirWindfield**
This commit adds support for specifying the target triple that blackjack
uses when generating the Bazel rules.

BREAKING CHANGE: The `-t,--target` argument is mandatory. Existing
tooling fails to run successfully due to it missing. To achieve the same
behavior as in v0.1.0, the target value should be set to
`x86_64-unknown-linux-gnu`.

Closes #5

### CODE REVIEW

**Comment by SirWindfield** [comment]
Superseeds #3

File: README.md [hunk 1]
```diff
@@ -103,6 +103,3 @@ prefix = "blackjack_crates_io"
 ```
 
 Now instead of `@crates_io_serde//:serde`, use `blackjack_crates_io_serde//:serde`.
-
-# Things that don't work yet (but would gladly accept a PR for)
-* Support for Windows and Mac. For the moment everything assumes your host and target is `x86_64-unknown-linux-gnu`
```
Comment (wildarch) [inline README.md:108]: Maybe we should leave a note here saying that Mac and Windows support is supported, but untested.

**Review by wildarch — CHANGES_REQUESTED** [review]
Thanks for putting this together, Sven!

I think this is good to go once we have a test to make sure the target flag is working as intended.

File: README.md (hunk 1 above)
Comment (gleyba) [reply README.md:108]: https://github.com/SirWindfield/blackjack/pull/1
some more progress 
works on mac

Comment (SirWindfield) [reply README.md:108]: I haven't had a lot of time lately for programming-related projects :( Would be nice if @wildarch could review the PR and see if it is more inline with the changes that he requested :smile:

Comment (wildarch) [reply README.md:108]: I've just reviewed the PR, looks great! 😄

File: tests/run.sh [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+echo | bazel run //:blackjack -- --target="$target"
 bazel run //:main
 popd
 
 pushd tests/workspace
-echo | bazel run //:blackjack
+echo | bazel run //:blackjack -- --target="$target"
 bazel run //crate1
 bazel run //crate2
 popd
```
[the remark this answers is not in the document]
Comment (wildarch) [reply tests/run.sh:24]: Actually, just going with Linux, Mac OS and Windows seems pretty easy to do in CI (https://github.com/wildarch/blackjack/pull/8 for reference), so maybe just use that for the tests?

[the remark this answers is not in the document]
Comment (wildarch) [reply tests/run.sh:24]: A simple crate to test with might be: https://docs.rs/os_info/3.0.0/os_info/index.html

**Comment by wildarch** [comment]
I'm going to close this PR now that #9 makes it obsolete. 

Thanks for all your efforts @SirWindfield! This was very helpful in putting #9 together 😄

**Comment by SirWindfield** [comment]
No problem! Glad that it worked out in the end! I might have some time for the Windows part :)

**Pull request closed without merging**